File: src/0242-valid-anagram/solution.py
from collections import Counter
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        string_s_count, string_t_count = Counter(s), Counter(t)

        # You can do Counter(t) == Counter(s) but this is the 
        # less abstracted version
        for k, v in string_s_count.items():
            if string_t_count[k] != v:
                return False
        return True

File: src/0242-valid-anagram/solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        // Bucket sort to allow for O(n) time complexity
        // But with O(26) Space, so this is most optimal
        int[] counts = new int[26];
        for (char c : s.toCharArray()) {
            counts[c - 'a']++;
        }
        // you can decrement or create another char array
        for (char c : t.toCharArray()) {
            counts[c - 'a']--;
        }

        for (int num : counts) {
            if (num != 0) {
                return false;
            }
        }
        return true;
    }
}
